node/src/neighborhood/mod.rs

```diff
@@ -732,19 +781,34 @@ impl Neighborhood {
     }
 
     fn handle_agrs(&mut self, agrs: Vec<AccessibleGossipRecord>, gossip_source: SocketAddr) {
-        let ignored_node_name = self.gossip_source_name(&agrs, gossip_source);
+        let gossip_source_name = self.gossip_source_name(&agrs, gossip_source);
         let gossip_record_count = agrs.len();
         let connection_progress_peers = self.overall_connection_status.get_peer_addrs();
         let acceptance_result = self
-            .gossip_acceptor_opt
+            .gossip_acceptor_info_opt
             .as_ref()
             .expect("Gossip Acceptor wasn't created.")
+            .as_ref()
+            .left_or_else (|_| panic! ("Neighborhood never got its PoolBindMessage"))
             .handle(
                 &mut self.neighborhood_database,
                 agrs,
                 gossip_source,
                 &connection_progress_peers,
             );
+        self.process_acceptance_result(
+            acceptance_result,
+            &gossip_source_name,
+            gossip_record_count,
+        );
+    }
+
+    fn process_acceptance_result(
+        &mut self,
+        acceptance_result: GossipAcceptanceResult,
+        gossip_source_name: &str,
```

Comment:
It's deliberately given an ambiguous name, because we don't know what it is at this point. Public keys are the most useful, so we look to see if we can find a public key on it. If we can, that's what gets put in the `gossip_source_name`. If we can't, we just use the `SocketAddr` for the `gossip_source_name`.